source/default_cmm.c

```diff
@@ -180,3 +185,11 @@ int aws_cryptosdk_default_cmm_set_alg_id(struct aws_cryptosdk_cmm *cmm, enum aws
     self->default_alg = alg_id;
     return AWS_OP_SUCCESS;
 }
+
+bool aws_cryptosdk_default_cmm_is_valid(const struct aws_cryptosdk_cmm *cmm) {
+    if (cmm == NULL) return false;
+
+    struct default_cmm *self = (struct default_cmm *)cmm;
+    return aws_cryptosdk_cmm_base_is_valid(&self->base) && aws_allocator_is_valid(self->alloc) &&
+           aws_cryptosdk_keyring_is_valid(self->kr);
+}
```

Comment:
Why this function wasn't written like...
```c
bool default_cmm_is_valid(const struct default_cmm *cmm) {
    return (cmm != NULL) &&
           aws_cryptosdk_cmm_base_is_valid(&cmm->base) &&
           aws_allocator_is_valid(cmm->alloc) &&
           aws_cryptosdk_keyring_is_valid(cmm->kr);
}
```
Shouldn't we validate both structures separately (i.e., on validity function for each data structure)?

Reply:
In general, extensions of `aws_cryptosdk_cmm` are always handled like this in the source code: The type is used as a parameter/return type and the struct is cast before doing any operations on it. This way, we are following API guidelines.

I am not sure what you mean by "validate both structures separately". We have `aws_cryptosdk_cmm_base_is_valid` for validating `aws_cryptosdk_cmm` structs and this one to validate `default_cmm` structs.